app/models/setting/discovered.rb

```diff
@@ -16,6 +16,7 @@ def self.load_defaults
     Setting.transaction do
       [
         self.set('discovery_fact', N_("Fact name to use for primary interface detection and hostname"), "discovery_bootif"),
+        self.set('discovery_hostname', N_("List of facts to use for the hostname (seperate by comma)"), "discovery_bootif"),
```

Comment:
> It doesn't accept empty value - empty value will trigger the exception at app/models/host/discovered.rb:46

Ok, if you want to allow blank *prefix*, then add it to the ALLOW_BLANK
constant. See Foreman Core for an example, remember to add a test for
that.

-- 
Later,
 Lukas #lzap Zapletal